frame/nomination-pools/src/lib.rs

```diff
@@ -582,19 +582,200 @@ pub struct PoolRoles<AccountId> {
+					t.change_rate.min_delay
+			}) {
+				return true
+			}
+			// check for `max_increase` throttling
+			(*to).saturating_sub(self.commission_or_zero()) > t.change_rate.max_increase &&
+				self.current.is_some()
+		} else {
+			return false
+		}
+	}
+
+	/// Set the pool's commission.
+	/// Update commission accordingly based on `commission` and `payee`.
+	/// If throttle is present, record the current block as the previously updated commission.
+	/// If the supplied commission is zero, `None` will be inserted and `payee` will be ignored.
+	fn maybe_update_current(
+		&mut self,
+		commission: &Perbill,
+		payee: T::AccountId,
+	) -> DispatchResult {
+		ensure!(!self.throttling(&commission), Error::<T>::CommissionChangeThrottled);
+		ensure!(
+			self.max.map(|m| commission <= &m).unwrap_or(true),
+			Error::<T>::CommissionExceedsMaximum
+		);
+
+		self.current = Some((*commission, payee));
+
+		// update `throttle.previous_set_at` if a throttle has been configured.
+		let _ = self
+			.throttle
+			.as_mut()
+			.map(|t| t.register_change(<frame_system::Pallet<T>>::block_number()));
+		Ok(())
+	}
+
+	/// Set the pool's maximum commission.
+	/// The pool's maximum commission can be set to any value initially, and only smaller values
+	/// thereafter. If larger values are attempted, this function will return a dispatch error.
+	///
+	/// If `current.0` is larger than an updated max commission value, then `current.0` will also be
+	/// updated to the new maximum.
+	fn maybe_update_max(&mut self, new_max: Perbill) -> DispatchResult {
+		if let Some(old) = self.max.as_mut() {
+			if new_max > *old {
+				return Err(Error::<T>::MaxCommissionRestricted.into())
+			}
+			*old = new_max;
+			let _ = self.current.as_mut().map(|(x, _)| *x = (*x).min(new_max));
+		} else {
+			self.max = Some(new_max)
+		};
+		Ok(())
+	}
+
+	/// Set the pool's commission throttle settings.
+	/// Once a throttle configuration has been set, only more restrictive values can be set
+	/// thereafter. These restrictions translate to increased `min_delay` values and decreased
+	/// `max_increase` values.
+	fn maybe_update_throttle(
+		&mut self,
+		change_rate: CommissionThrottlePrefs<T::BlockNumber>,
+	) -> DispatchResult {
+		if let Some(t) = &self.throttle {
+			ensure!(!t.restricted(&change_rate), Error::<T>::CommissionThrottleNotAllowed);
+		}
+		self.throttle = self
+			.throttle
+			.as_ref()
+			.map_or(Some(CommissionThrottle { change_rate, previous_set_at: None }), |t| {
+				Some(CommissionThrottle { change_rate, ..*t })
+			});
+		Ok(())
+	}
+
+	/// Gets the current commission (if any) and payee to be paid.
+	/// A zero commission along with a `None` payee is returned in the event a commission has not
+	/// been configured to the pool.
+	fn get_commission_and_payee(
+		&self,
+		pending_rewards: &BalanceOf<T>,
+	) -> Option<(BalanceOf<T>, T::AccountId)> {
+		self.current
+			.as_ref()
+			.map(|(commission, payee)| (*commission * *pending_rewards, payee.clone()))
```

Comment:
```suggestion
			.map(|(commission, payee)| (*commission.saturating_mul(*pending_rewards), payee.clone()))
```

Reply:
If we use `saturating_mul` here it looks like we need to introduce `PerbillToBalance::convert()`, but `From<sp_runtime::Perbill>` is not implemented for `u128. Is it worth adding?

Reply:
Discussed and do not need `saturating_mul`.